src/model.py

```diff
@@ -33,9 +41,23 @@ def build_model_and_tokenizer_for(
     ]
 
     logger.info(f"Loading the {model_name} model")
-    model = transformers.AutoModelForCausalLM.from_pretrained(
-        model_name, bad_words_ids=bad_words_ids)
-    model.eval().half().to("cuda")
+    # If loading 6B model in, we need to manually download files and then use the accelerate library.
+    if model_name == "PygmalionAI/pygmalion-6b":
+        assert USE_ACCELERATE, "Accelerate must be installed in order to use Pygmalion-6B!"
+        index_file = _download_hf_files(model_name)
+        config = transformers.AutoConfig.from_pretrained(model_name)
+        # Manually add bad words
+        config.bad_words_ids = bad_words_ids
+        
+        with accelerate.init_empty_weights():
+            model = transformers.AutoModelForCausalLM.from_config(config)
+        model = accelerate.load_checkpoint_and_dispatch(
+            model, index_file, device_map="auto", no_split_module_classes=["GPTJBlock"])
+        model.eval().half()
+    else:
+        model = transformers.AutoModelForCausalLM.from_pretrained(
+            model_name, bad_words_ids=bad_words_ids)
+        model.eval().half().to("cuda")
```

Comment:
How do you feel about this instead?

```suggestion
    if USE_ACCELERATE:
        index_file = _download_hf_files(model_name)
        config = transformers.AutoConfig.from_pretrained(model_name)
        # Manually add bad words
        config.bad_words_ids = bad_words_ids
        
        with accelerate.init_empty_weights():
            model = transformers.AutoModelForCausalLM.from_config(config)
        model = accelerate.load_checkpoint_and_dispatch(
            model, index_file, device_map="auto", no_split_module_classes=["GPTJBlock"])
        model.eval().half()
    else:
        model = transformers.AutoModelForCausalLM.from_pretrained(
            model_name, bad_words_ids=bad_words_ids)
        model.eval().half().to("cuda")
```

That way the optimized loading happens independently of which model is being used. Or does that have some negative side-effect that we'd like to avoid on the smaller models?

Reply:
The smaller models right now do not have an index.json file nor sharded models, meaning that would throw an error until we add index.json files to all previous model repos and then refactor the _download_hf_file function to generalize it. I had an idea for how to do that in the comments of that function, but you might need to add the index.json files yourself

Reply:
Ah, I see. I'll take a quick gander through the HF code to see how they determine which files to download and see whether we can easily import and use that here, then I'll get back to you.

In the meantime though - feel free to point the `git clone` in the example notebook at this branch so everything works while we iron this out, if you'd like.

Reply:
Got it.